src/platform/user/authentication/components/SessionTimeoutModal.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import moment from 'moment';
+
+import Modal from '@department-of-veterans-affairs/formation-react/Modal';
+
+import recordEvent from 'platform/monitoring/record-event';
+import { logout } from 'platform/user/authentication/utilities';
+import { teardownProfileSession } from 'platform/user/profile/utilities';
+import localStorage from 'platform/utilities/storage/localStorage';
+
+const MODAL_DURATION = 30; // seconds
+
+class SessionTimeoutModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { countdown: null };
+    this.expirationInterval = null;
+  }
+
+  componentDidUpdate() {
+    if (this.props.isLoggedIn && !this.expirationInterval) {
+      this.clearInterval();
+      this.expirationInterval = setInterval(this.checkExpiration, 1000);
+    }
+  }
+
+  componentWillUnmount() {
+    this.clearInterval();
+  }
+
+  clearInterval = () => {
+    clearInterval(this.expirationInterval);
+    this.expirationInterval = null;
+  };
+
+  checkExpiration = () => {
+    if (!this.props.isLoggedIn) {
+      this.clearInterval();
+      return;
+    }
+
+    const expirationDate = localStorage.getItem('sessionExpiration');
+    if (!expirationDate) return;
+
+    const expirationMoment = moment(new Date(expirationDate).toISOString());
```

Comment:
If we're just comparing seconds, this can probably all be done w/ the built-in JS date methods. e.g `if (Date.now() - expirationDate < 0)...`, but not a big deal to use moment for readability and if we're using it elsewhere.

Reply:
Just realizing this would also require the expirationDate be sent from the backend in the same format (milliseconds since epoch).  Probably not worth the effort.

Reply:
That shouldn't be a problem, since we can do `new Date(expirationDate).getTime()`. I think native will be better for performance, so I'll try making this change.